src/Parser.js

```diff
@@ -2,25 +2,28 @@ import {COMMENT, RULESET, DECLARATION} from './Enum.js'
 import {abs, trim, from, sizeof, strlen, substr, append, replace} from './Utility.js'
 import {node, char, next, peek, caret, alloc, dealloc, delimit, whitespace, identifier, commenter} from './Tokenizer.js'
 
+/** @typedef {import('./Middleware.js').Element} Element */
+/** @typedef {string[]} Rulesets */
```

Comment:
IMHO it's better to declare `Ruleset` and use `Ruleset[]` later. But anyway - I'm not sure if this is declared correctly, this type is used as a return type of the `compile` which returns `Element[]` - it's an array of objects, not an array of strings

Reply:
OK good to know - it's not clear that `compile` is returning `Element[]` because `compile` is `dealloc` (which returns what it is given) after a `parse` (which ends in `return rulesets`). Are you suggesting that `rulesets` in `parse` is `Element[]`?

I like the note about using `Ruleset` and `Ruleset[]`. I'll do that thanks